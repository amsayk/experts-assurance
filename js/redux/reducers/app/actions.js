import {
  RESIZE,
  CONNECTION_STATE_CHANGE,
  APP_STATE_CHANGE,
  READY,
  TOGGLE_ALERTS,

  ADD_DOC,
} from './constants';

export function resize() {
  return {
    type: RESIZE,
  };
}

export function connectionStateChange(isConnected) {
  return {
    type: CONNECTION_STATE_CHANGE,
    isConnected,
  };
}

export function appStateChange(currentState) {
  return {
    type: APP_STATE_CHANGE,
    currentState,
  };
}

export function ready() {
  return {
    type: READY,
  };
}

export function toggleAlerts() {
  return {
    type: TOGGLE_ALERTS,
  };
}

export function startAddingDoc() {
  return {
    type      : ADD_DOC,
    addingDoc : true,
  };
}

export function finishAddingDoc() {
  return {
    type      : ADD_DOC,
    addingDoc : false,
  };
}

export function saveNewDoc(data) {
  return async (dispatch, _, { client }) => {

  };
}

