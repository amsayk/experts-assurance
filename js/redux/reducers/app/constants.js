import keyOf from 'keyOf';

export const READY = keyOf({READY: null});
export const RESIZE = keyOf({RESIZE: null});
export const CONNECTION_STATE_CHANGE = keyOf({CONNECTION_STATE_CHANGE: null});

