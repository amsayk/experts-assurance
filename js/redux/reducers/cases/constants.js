import keyOf from 'keyOf';

import {
  ACTION as SORT_ACTION,
} from 'redux/reducers/sorting/constants';

import {
  ACTION as SELECTION_ACTION,
} from 'redux/reducers/selection/constants';

// ------------------------------------
// Constants
// ------------------------------------

export const ON_SEARCH = keyOf({CASES_ON_SEARCH: null});
export const ON_STATE = keyOf({CASES_ON_STATE: null});
export const ON_CLIENT = keyOf({CASES_ON_CLIENT: null});
export const ON_INSURER = keyOf({CASES_ON_INSURER: null});

export const SORT = keyOf({[`${SORT_ACTION}/cases`]: null});
export const SELECTION = keyOf({[`${SELECTION_ACTION}/cases`]: null});

