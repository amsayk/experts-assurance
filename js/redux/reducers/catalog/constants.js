import keyOf from 'keyOf';

import {
  ACTION as SORT_ACTION,
} from 'redux/reducers/sorting/constants';

import {
  ACTION as SELECTION_ACTION,
} from 'redux/reducers/selection/constants';

// ------------------------------------
// Constants
// ------------------------------------

// view
export const VIEW_TYPE_LIST = keyOf({CATALOG_VIEW_TYPE_LIST: null});
export const VIEW_TYPE_GRID = keyOf({CATALOG_VIEW_TYPE_GRID: null});

export const START_ADDING = keyOf({CATALOG_START_ADDING: null});
export const STOP_ADDING  = keyOf({CATALOG_STOP_ADDING: null});

export const TOGGLE_SEARCH = keyOf({CATALOG_TOGGLE_SEARCH: null});
export const TOGGLE_SIDE_MENU = keyOf({CATALOG_TOGGLE_SIDE_MENU: null});

export const SORT = keyOf({[`${SORT_ACTION}/catalog`]: null});
export const SELECTION = keyOf({[`${SELECTION_ACTION}/catalog`]: null});

