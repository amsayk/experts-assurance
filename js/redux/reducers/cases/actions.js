import {
  ON_SEARCH,
  ON_STATE,
  ON_CLIENT,
  ON_MANAGER,
  ON_AGENT,
} from './constants';

// Selection
import { set, add, toggle } from 'redux/reducers/selection/actions';
export const toggleSelection = toggle('cases');
export const setSelection    = set('cases');
export const addToSelection  = add('cases');

// Sorting
import { sort, sortKey, sortDirection } from 'redux/reducers/sorting/actions';
export const sortCases            = sort('cases');
export const sortCasesByKey       = sortKey('cases');
export const sortCasesByDirection = sortDirection('cases');

export function search(queryString) {
  return {
    type : ON_SEARCH,
    queryString,
  };
}

export function onClient(id) {
  return {
    type : ON_CLIENT,
    id,
  };
}

export function onManager(id) {
  return {
    type : ON_MANAGER,
    id,
  };
}

export function onAgent(id) {
  return {
    type : ON_AGENT,
    id,
  };
}

export function onState(state) {
  return {
    type : ON_STATE,
    state,
  };
}

