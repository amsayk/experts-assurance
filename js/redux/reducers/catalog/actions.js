import {
  VIEW_TYPE_GRID,
  VIEW_TYPE_LIST,
  START_ADDING,
  STOP_ADDING,
  TOGGLE_SEARCH,
  TOGGLE_SIDE_MENU,
} from './constants';

// Selection
import { set, add } from 'redux/reducers/selection/actions';
export const setSelection = set('catalog');
export const addToSelection = add('catalog');

// Sorting
import { sort, sortKey, sortDirection } from 'redux/reducers/sorting/actions';
export const sortProducts = sort('catalog');
export const sortProductsByKey = sortKey('catalog');
export const sortProductsByDirection = sortDirection('catalog');

export function viewTypeGrid() {
  return {
    type: VIEW_TYPE_GRID,
  };
}
export function viewTypeList() {
  return {
    type: VIEW_TYPE_LIST,
  };
}

export function toggleSearch() {
  return {
    type: TOGGLE_SEARCH,
  };
}

export function startAdding() {
  return {
    type: START_ADDING,
  };
}

export function stopAdding() {
  return {
    type: STOP_ADDING,
  };
}

export function toggleSideMenu() {
  return {
    type : TOGGLE_SIDE_MENU,
  };
}

