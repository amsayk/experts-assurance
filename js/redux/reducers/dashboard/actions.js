import {
  TOGGLE,
  SET_CATEGORY,
  SET_DURATION,
  // TOGGLE_INCLUDE_CANCELED,
  // SET_ONLY_VALID_OPEN,
} from './constants';

// Sorting
import { sort } from 'redux/reducers/sorting/actions';

export const sortOpen       = sort('openDashboard');
// export const sortClosed  = sort('closedDashboard');
export const sortUnpaid     = sort('unpaidDashboard');
export const sortInvalid    = sort('invalidDashboard');

export function toggle(dashboard) {
  return {
    type : TOGGLE,
    dashboard,
  };
}

export function setDuration(dashboard, duration) {
  return {
    type : SET_DURATION,
    dashboard,
    duration,
  };
}

export function setCategory(category) {
  return {
    type :  SET_CATEGORY,
    dashboard : 'invalid',
    category,
  };
}

// export function toggleIncludeCanceled() {
//   return {
//     type : TOGGLE_INCLUDE_CANCELED,
//   };
// }

// export function onOnlyValidOpenChange(onlyValidOpen) {
//   return {
//     type : SET_ONLY_VALID_OPEN,
//     onlyValidOpen,
//   };
// }

