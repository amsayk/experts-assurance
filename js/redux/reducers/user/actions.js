import Parse from 'parse';
import { USER_LOGGED_IN, USER_LOGGED_OUT } from './constants';

import debug from 'log';

const log = debug('app:client:auth');

export function login(payload) {
  return {
    type: USER_LOGGED_IN,
    payload,
  };
}

export function logOut(manual = true) {
  return async (dispatch, _, { client }) => {
    try {
      await Parse.User.logOut();
    } catch (e) {
      switch (e.code) {
        case Parse.Error.INVALID_SESSION_TOKEN:
          log.error('INVALID_SESSION_TOKEN: clearing storage.');
          return Parse.CoreManager.getStorageController().clear();
        default:
          log.error(e);
      }
    } finally {
      dispatch({ type: USER_LOGGED_OUT, manual });
      // client.resetStore();
    }
  };
}

