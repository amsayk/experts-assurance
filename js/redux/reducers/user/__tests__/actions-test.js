import { login, logOut } from '../actions';
import { USER_LOGGED_IN, USER_LOGGED_OUT } from '../constants';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';

const history = {
  push: jest.fn(),
};

const client = {
  resetStore: jest.fn(),
};

const middlewares = [thunk.withExtraArgument({history, client})];
const mockStore = configureStore(middlewares);

describe('user actions', () => {
  it('login', () => {
    const payload = {};
    const action = login(payload);

    expect(action.type).toBe(USER_LOGGED_IN);
    expect(action.payload).toBe(payload);

    expect(action).toMatchSnapshot();
  });

  it('logOut should reset store and redirect', async () => {
    const store = mockStore({});
    try {
      await store.dispatch(logOut());
    } catch (e) {}
    const actions = store.getActions();
    expect(actions[0].type).toBe(USER_LOGGED_OUT);
    expect(client.resetStore).toHaveBeenCalled();
    expect(history.push).toBeCalledWith('/');
  });
});

