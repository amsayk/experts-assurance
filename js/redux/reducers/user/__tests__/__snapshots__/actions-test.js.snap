exports[`user actions login 1`] = `
Object {
  "payload": Object {},
  "type": "USER_LOGGED_IN",
}
`;
